manager/network/model.go

```diff
@@ -0,0 +1,15 @@
+package network
+
+import (
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/manager/allocator/networkallocator"
+)
+
+// Model is an abstraction over the Network Model to be used.
+type Model interface {
+	Allocator() (networkallocator.NetworkAllocator, error)
+	SetDefaults(spec *api.NetworkSpec) error
+	ValidateNetworkSpec(spec *api.NetworkSpec) error
+	PredefinedNetworks() []networkallocator.PredefinedNetworkData
```

Comment:
Actually `PredefinedNetworkData` isn't used anywhere in the network allocator package after the changes in this PR so moving it makes absolute sense and doesn't cause a circular dependency.

I'm still considering possibly moving `networkallocator.NetworkAllocator` to `network.Allocator`, I think I like it though.